lib/etcd_lib.py

```diff
@@ -47,6 +47,26 @@ def get_ingress_address(endpoint_name):
     return sorted(all_addrs, key=lambda i: ip_address(i).version)[0]
 
 
+def get_snapshot_count(snapshot_count: str, channel: str) -> int:
+    """Returns the snapshot count value
+
+    * check if the value is auto,
+        iff channel >=3.2 it will set 100'000 otherwhise it will set 10'000
+    * any other integer value will be set as it is
+
+    @param snapshot_count the value to set, could be a number or auto
+    @param channel the channel used by the charm
+    """
+    if snapshot_count == "auto":
+        if channel == "auto" or float(channel.split("/")[0]) >= 3.2:
+            return 100000
+        return 10000
+    try:
+        return int(snapshot_count)
+    except ValueError:
+        raise TypeError(f"{snapshot_count} value is not an integer number")
```

Comment:
Hopefully we never reach etcd  - 3.10/stable

```suggestion
    SNAPSHOT_COUNT_PRIOR_32 = 10000
    SNAPSHOT_COUNT_BEYOND_32 = 100000

    if snapshot_count == "auto":
        if channel == "auto":
            return SNAPSHOT_COUNT_BEYOND_32
        maj, min = map(int, c.split("/")[0].split("."))
        if (maj, min) >= (3, 2)
            return SNAPSHOT_COUNT_BEYOND_32
        return SNAPSHOT_COUNT_PRIOR_32
    try:
        return int(snapshot_count)
    except ValueError:
        raise TypeError(f"{snapshot_count} value is not an integer number")
```
```

Reply:
That's a really good point. We can also use `Version` for that matter.
```python
from packaging.version import Version
Version("2.3.1") > Version("10.1.2")  # False
```

Reply:
@HomayoonAlimohammadi TIL, thanks